agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/ConfigurationBuilder.java

```diff
@@ -233,6 +235,15 @@ public static void overlayEnvVars(Configuration config) throws IOException {
         loadInstrumentationEnabledEnvVars(config);
     }
 
+    private static void overlayRpConfiguration(Configuration config, RpConfiguration rpConfiguration)  {
+        if (rpConfiguration.connectionString != null) {
+            config.connectionString = rpConfiguration.connectionString;
+        }
+        if (rpConfiguration.sampling != null) {
```

Comment:
if (rpconfiguration.sampling != null && config.sampling.percentage != rpConfiguration.sampling.percentage) {
only update when it's different?

Reply:
I don't think any reason not to update the value when it's the same here